client_configuration.go

```diff
@@ -129,14 +164,29 @@ func validateClientConfiguration(clientConfig *ClientConfig) error {
 func validateLogLevel(clientConfig ClientConfig) error {
 	var logLevel = clientConfig.Common.LogLevel
 	if logLevel != "" {
-		_, error := toLogLevel(logLevel)
-		if error != nil {
-			return error
+		_, err := toLogLevel(logLevel)
+		if err != nil {
+			return err
 		}
 	}
 	return nil
 }
 
+func isCfgPermValid(filePath string) (bool, error) {
+	if runtime.GOOS != "windows" {
+		stat, err := os.Stat(filePath)
+		if err != nil {
+			return false, err
+		}
+		perm := stat.Mode()
+		// Check if group (5th LSB) or others (2nd LSB) have a write permission to the file
+		if perm&(1<<4) != 0 || perm&(1<<1) != 0 {
```

Comment:
1<<4 is 16, 1 << 1 is 2. Is this "<<" form helpful for any reason?

Reply:
In my opinion it is easier to understand in this case, as we check the bit on a certain position responsible for a specific access. E.g. 1<<4 means that we check fifth bit.